### PR TITLE
Choose lowest downsampling resolution with granular control.

### PR DESCRIPTION
**Pull request by bwplotka**
Playing with `Select` `storage.SelectParams.Step`

As a follow up on this: https://github.com/thanos-io/thanos/pull/1431 I was experimenting with subquery + resolution in order to automatically adjust what downsampling Thanos should use. This, in theory, sounds great but looks like I don't really understand PromQL `storage.Params.Step` field properly. Need to investigate more (is this param not propagated)?

TODO: 
* [] Proper design
* [] Investigate why the step is not propagated from `[...:<resolution>]`

Signed-off-by: Bartek Plotka <bwplotka@gmail.com>

### CODE REVIEW

File: pkg/query/querier_test.go [hunk 1]
```diff
@@ -22,23 +22,180 @@ import (
...
+							Name:  "__name__",
+							Value: "up",
+						},
+					},
+					Aggregates: []storepb.Aggr{storepb.Aggr_COUNT, storepb.Aggr_SUM},
+				},
+			},
+		},
+		{
+			query: "rate(up[5m:200s])",
```
Comment (bwplotka) [inline pkg/query/querier_test.go]: This does not work as expected (in my mind). Step is still passed as 100 ms no matter what. cc @brian-brazil for help =D

Comment (bwplotka) [reply pkg/query/querier_test.go]: Based my knowledge on: https://prometheus.io/blog/2019/01/28/subquery-support/#subqueries

Comment (brian-brazil) [reply pkg/query/querier_test.go]: The step is the one for query_range. We're not passing any hints relating to subqueries currently.

Comment (bwplotka) [reply pkg/query/querier_test.go]: Thanks! Will look deeper in PromQL if it's sane to actually propagate that hint & how.

**Comment by damnever** [comment]
The step in query hints seems to be fixed now. The remaining issue is that we still can’t tune lookback delta for subqueries.

**Pull request closed without merging**